packages/@sanity/form-builder/src/inputs/files/ImageInput/ImageInput.tsx

```diff
@@ -552,30 +558,41 @@ export default class ImageInput extends React.PureComponent<Props, ImageInputSta
 
     return (
       <WithReferencedAsset observeAsset={observeAsset} reference={asset}>
-        {(assetDocument) => (
-          <ImageActionsMenu
-            isMenuOpen={isMenuOpen}
-            onEdit={this.handleOpenDialog}
-            showEdit={showAdvancedEditButton}
-            onMenuOpen={(isOpen) => this.setState({isMenuOpen: isOpen})}
-          >
-            <ActionsMenu
-              onUpload={this.handleSelectFiles}
-              browse={browseMenuItem}
-              onReset={this.handleRemoveButtonClick}
-              downloadUrl={imageUrlBuilder
-                .image(value.asset)
-                .forceDownload(
-                  assetDocument.originalFilename || `download.${assetDocument.extension}`
-                )
-                .url()}
-              copyUrl={imageUrlBuilder.image(value.asset).url()}
-              readOnly={readOnly}
-              directUploads={directUploads}
-              accept={accept}
-            />
-          </ImageActionsMenu>
-        )}
+        {(assetDocument) => {
+          let copyUrl: string | undefined
+          let downloadUrl: string | undefined
+
+          if (isImageSource(value)) {
+            const filename = assetDocument.originalFilename || `download.${assetDocument.extension}`
+            downloadUrl = imageUrlBuilder.image(assetDocument._id).forceDownload(filename).url()
+            copyUrl = imageUrlBuilder.image(assetDocument._id).url()
+          }
+
+          return (
+            <ImageActionsMenu
```

Comment:
This might be more of a question for @rexxars since I see that this was also changed in the v3 version. But why did we split the menus specifically based on the image / file input? The menu (at least the one I recall) was with the same behavior?

Reply:
read @sjelfull earlier comment and I confused two different menus that are in the image input (confused `ImageActionsMenu` with `ActionsMenu` 🥲 ) nevermind this question ✨